sdk/services/kwentaToken.ts

```diff
@@ -0,0 +1,464 @@
+					vestable,
+					amount: Number(ethers.utils.formatEther(next.escrowAmount)),
+					fee: Number(ethers.utils.formatEther(vestingEntries[i].fee)),
+					status: date > Date.now() ? 'VESTING' : 'VESTED',
+				});
+
+				return acc;
+			},
+			{ escrowData: [] as EscrowData[], totalVestable: 0 }
+		);
+
+		return { escrowData, totalVestable };
+	}
+
+	public getReward() {
+		const { KwentaStakingRewards } = this.sdk.context.contracts;
+
+		if (!KwentaStakingRewards) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		return this.sdk.transactions.createContractTxn(KwentaStakingRewards, 'getReward', []);
+	}
+
+	// TODO: Delete `approveLPToken` method.
+	// In that case, we can safely remove the map object from this method.
+
+	public approveKwentaToken(
+		token: 'kwenta' | 'vKwenta' | 'veKwenta',
+		amount = ethers.constants.MaxUint256
+	) {
+		const {
+			KwentaToken,
+			KwentaStakingRewards,
+			vKwentaToken,
+			vKwentaRedeemer,
+			veKwentaToken,
+			veKwentaRedeemer,
+		} = this.sdk.context.contracts;
+
+		const map = {
+			kwenta: { contract: KwentaToken, spender: KwentaStakingRewards },
+			vKwenta: { contract: vKwentaToken, spender: vKwentaRedeemer },
+			veKwenta: { contract: veKwentaToken, spender: veKwentaRedeemer },
+		};
+
+		const { contract, spender } = map[token];
+
+		if (!contract || !spender) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		return this.sdk.transactions.createContractTxn(contract, 'approve', [spender.address, amount]);
+	}
+
+	public approveToken(
+		token: ContractName,
+		spender?: ContractName,
+		amount = ethers.constants.MaxUint256
+	) {
+		const tokenContract = this.sdk.context.contracts[token];
+
+		if (!tokenContract) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		let spenderAddress = this.sdk.context.walletAddress;
+
+		if (spender) {
+			const spenderContract = this.sdk.context.contracts[spender];
+			if (spenderContract) spenderAddress = spenderContract.address;
+		}
+
+		return this.sdk.transactions.createContractTxn(tokenContract, 'approve', [
+			spenderAddress,
+			amount,
+		]);
+	}
+
+	public redeemToken(token: ContractName) {
+		const tokenContract = this.sdk.context.contracts[token];
+
+		if (!tokenContract) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		return this.sdk.transactions.createContractTxn(tokenContract, 'redeem', [
+			this.sdk.context.walletAddress,
+		]);
+	}
+
+	public redeemVKwenta() {
+		return this.redeemToken('vKwentaRedeemer');
+	}
+
+	public redeemVeKwenta() {
+		return this.redeemToken('veKwentaRedeemer');
+	}
+
+	public vestToken(ids: number[]) {
+		const { RewardEscrow } = this.sdk.context.contracts;
+
+		if (!RewardEscrow) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		return this.sdk.transactions.createContractTxn(RewardEscrow, 'vest', [ids]);
+	}
+
+	public stakeKwenta(amount: string | ethers.BigNumber) {
```

Comment:
Yeah it's not necessary. It still returns a promise. `async` is only necessary when there's an await in the function body.

Reply:
ok yea just more a readability thing then, would be good to either add it here or remove from the others for consistency.

Reply:
The thing is, those other instances require the `async` keyword since we use `await` in their body. However, here, we don't use `await` in the body, so we can omit both the `async` and `await` and still return a promise.

Reply:
Ah sorry I just mean specifically these functions all calling  `this.performStakeAction`, they all have async apart from this one, so just a clean up thing for consistency:
<img width="598" alt="Screenshot 2022-12-05 at 07 48 25" src="https://user-images.githubusercontent.com/3802342/205582425-eda3832e-98b3-4010-a24d-bf92828e19fe.png">

Reply:
Ah. Yeah. Good catch. For some reason, I didn't notice that those were still using `async`.